### PR TITLE
Add skip_retries and skip_scheduled options to Sidekiq macro

### PR DESCRIPTION
**Pull request by prsimp**
First, thanks for your work on HireFire!

With very large scheduled and retry queues, the enumeration over the queues to count relevant jobs can cause very long requests. On platforms like Heroku, this will cause the request to timeout before being completed, thus creating checkup issues for the app.

Let me know if you think there's a better way to approach this. It didn't seem like there was a more performant way of calculating these sums, so I opted to skip them. While not being able to pre-spin-up workers for scheduled or soon-to-be-retried jobs is less than ideal, it is certainly favorable to app-wide checkup issues.

### CODE REVIEW

**Comment by meskyanichi** [comment]
Hi @prsimp,

Thanks for the PR!

Adding this functionality seems fine to me.

I see that you've added two accessors, as well as the `options` variable. However, you're currently not using the `options` variable.

I think it'd be better to remove the two accessors and use the `options` variable that you've defined instead to configure whether or not to skip retries/scheduled.

i.e. instead of `skip_retries` and `skip_scheduled`, use `options[:skip_retries]` and `options[:skip_scheduled]`. This way the configuration isn't global and you can have different options for different queues if necessary.

If you can make that change, and amend/rebase, I'll pull it in. :)

File: lib/hirefire/macro/sidekiq.rb [hunk 1]
```diff
@@ -5,6 +5,14 @@ module Macro
     module Sidekiq
       extend self
 
+      # @return [Boolean] Determines whether scheduled jobs should be counted.
+      #
+      attr_accessor :skip_scheduled
```
Comment (meskyanichi) [inline lib/hirefire/macro/sidekiq.rb]: remove this

File: lib/hirefire/macro/sidekiq.rb [hunk 2]
```diff
@@ -5,6 +5,14 @@ module Macro
...
+
+      # @return [Boolean] Determines whether retries should be counted.
+      #
+      attr_accessor :skip_retries
```
Comment (meskyanichi) [inline lib/hirefire/macro/sidekiq.rb]: remove this

File: lib/hirefire/macro/sidekiq.rb [hunk 3]
```diff
@@ -26,14 +42,18 @@ def queue(*queues)
           memo
         end
 
-        in_schedule = ::Sidekiq::ScheduledSet.new.inject(0) do |memo, job|
-          memo += 1 if queues.include?(job["queue"]) && job.at <= Time.now
-          memo
+        if !skip_scheduled
```
Comment (meskyanichi) [inline lib/hirefire/macro/sidekiq.rb]: change to `if !options[:skip_scheduled]`

File: lib/hirefire/macro/sidekiq.rb [hunk 4]
```diff
@@ -26,14 +42,18 @@ def queue(*queues)
...
+          in_schedule = ::Sidekiq::ScheduledSet.new.inject(0) do |memo, job|
+            memo += 1 if queues.include?(job["queue"]) && job.at <= Time.now
+            memo
+          end
         end
 
-        in_retry = ::Sidekiq::RetrySet.new.inject(0) do |memo, job|
-          memo += 1 if queues.include?(job["queue"]) && job.at <= Time.now
-          memo
+        if !skip_retries
```
Comment (meskyanichi) [inline lib/hirefire/macro/sidekiq.rb]: change to `if !options[:skip_retries]`

**Comment by meskyanichi** [comment]
Then you can simply use:

```rb
HireFire::Macro::Sidekiq.queue(skip_retries: true, skip_scheduled: true)
HireFire::Macro::Sidekiq.queue("mailer", skip_retries: true, skip_scheduled: true)
# etc
```

If you could also add a comment with an example using these options that'd be great!

**Comment by prsimp** [comment]
@meskyanichi -- I debated on which implementation I liked better, but ended up going with global setting as the first pass and forgot to remove some of the options code. In the end, it just made the switch to an options-based implementation even easier ;)

I've added examples as well. Thanks for the quick feedback and let me know if there's anything else I can do!

**Comment by meskyanichi** [comment]
Thanks @prsimp! I've merged your commit and released v0.3.9. :+1:

**Pull request closed without merging**